lib/Castle/Errors.php

```diff
@@ -54,3 +54,8 @@ class Castle_InvalidParametersError extends Castle_ApiError
 {
 
 }
+
+class Castle_InvalidRequestTokenError extends Castle_InvalidParametersError
```

Comment:
Why do we extend our own Errors ? What is the purpose of this ? I think we should always extend `Exception` class as there is no need of having a tree of exception classes inheritance.

Reply:
The invalid token error is a subtype of invalid parameter error, so if someone only handles `Castle_InvalidParametersError` in their code, we want the invalid token error to be part of that, unless handled specifically.

Reply:
Ah so we don't want to break things. Logical 👍